### PR TITLE
Adopt MavenArchiver Plugin according to CQ 24115

### PR DESCRIPTION
**Pull request by laeubi**
CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=24115
Squashed commits up to https://github.com/mickaelistria/m2e-core/commit/e4e68c7

### CODE REVIEW

**Comment by laeubi** [comment]
Build is green, so this is ready to be merged...

**Review by akurtakov — APPROVED** [review]

File: org.eclipse.m2e.mavenarchiver/.settings/org.eclipse.jdt.core.prefs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+eclipse.preferences.version=1
+org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
+org.eclipse.jdt.core.compiler.codegen.methodParameters=do not generate
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=11
```
Comment (HannesWell) [inline org.eclipse.m2e.mavenarchiver/.settings/org.eclipse.jdt.core.prefs:4]: Has to be set to Java-17. Ideally you just set it via the Right-Click -> Build Path -> Configure Build Path and then set the JRE System Library. This also updates some other settings.

File: org.eclipse.m2e.mavenarchiver/.classpath [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
+	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
```
Comment (HannesWell) [inline org.eclipse.m2e.mavenarchiver/.classpath:5]:
```suggestion
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17">
		<attributes>
			<attribute name="module" value="true"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
	<classpathentry kind="src" path="src"/>
```
Update to Java-17 and unify with other projects

File: org.eclipse.m2e.mavenarchiver/.project [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.m2e.core.maven2Builder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
```
Comment (HannesWell) [inline org.eclipse.m2e.mavenarchiver/.project:17]:
```suggestion
```
Can be deleted. No more pom in this project.

File: org.eclipse.m2e.mavenarchiver/.project [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+		<buildCommand>
+			<name>org.eclipse.pde.SchemaBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.m2e.core.maven2Nature</nature>
```
Comment (HannesWell) [inline org.eclipse.m2e.mavenarchiver/.project:30]:
```suggestion
```
Can be deleted. No more pom in this project.

File: org.eclipse.m2e.mavenarchiver/.settings/org.eclipse.m2e.core.prefs [hunk 5]
```diff
@@ -0,0 +1,4 @@
+activeProfiles=
```
Comment (HannesWell) [inline org.eclipse.m2e.mavenarchiver/.settings/org.eclipse.m2e.core.prefs:1]: Entire file can be deleted if the m2e builder/nature is removed.

File: org.eclipse.m2e.mavenarchiver/META-INF/MANIFEST.MF [hunk 6]
```diff
@@ -0,0 +1,15 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: m2e connector for the mavenarchiver and pom properties
+Bundle-SymbolicName: org.eclipse.m2e.mavenarchiver;singleton:=true
+Bundle-Version: 2.0.0.qualifier
+Bundle-Vendor: Eclipse m2e
```
Comment (HannesWell) [inline org.eclipse.m2e.mavenarchiver/META-INF/MANIFEST.MF:6]:
```suggestion
Bundle-Vendor: Eclipse.org - m2e
```
To be consistent wit other m2e projects.

**Review by HannesWell — APPROVED** [review]
Thank you Christoph.

A few configuration remarks regarding project configuration but no blockers.

**Comment by HannesWell** [comment]
Looks like this took too long. I will open a follow up.

**Comment by HannesWell** [comment]
Btw. @laeubi are you preparing a PR for the tests (I suspect in tesla/m2e-core-tests) already?

**Comment by laeubi** [comment]
I'll try to do that, but If you want I won't mind to take over there ;-)

**Pull request merged**